src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,32 +34,62 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        } else {
+            Scanner addressScanner = new Scanner(trimmedAddress).useDelimiter(ADDRESS_DELIMITER);
+            setAddress(addressScanner.next(), addressScanner.next(), addressScanner.next(), addressScanner.next());
```

Comment:
If one doesn't add the comma, the validation would have failed, thus the code will not execute this line. (: